tests/test_ingest.py

```diff
@@ -25,38 +25,71 @@
 import tempfile
 import unittest
 
-import lsst.daf.butler as dafButler
 import lsst.daf.butler.tests as butlerTests
+from lsst.daf.butler import DatasetType, Butler, DataCoordinate
+from lsst.daf.butler.core.utils import getFullTypeName
 
+from lsst.obs.base.ingest_tests import IngestTestBase
 from lsst.obs.base import RawIngestTask
 
 
 TESTDIR = os.path.dirname(__file__)
 
 
-class RawIngestTestCase(unittest.TestCase):
+class DummyCamRawIngestTask(RawIngestTask):
+    """For DummyCam we ingest a different dataset type that can return
+    a non-Exposure"""
+
+    def getDatasetType(self):
+        """Return the DatasetType of the datasets ingested by this Task.
+        """
+        return DatasetType("raw_dict", ("instrument", "detector", "exposure"), "StructuredDataDict",
+                           universe=self.butler.registry.dimensions)
+
+
+class RawIngestTestCase(IngestTestBase, unittest.TestCase):
+
+    ingestDatasetTypeName = "raw_dict"
+    rawIngestTask = getFullTypeName(DummyCamRawIngestTask)
+    curatedCalibrationDatasetTypes = ()
+    ingestDir = TESTDIR
+    instrumentClassName = "lsst.obs.base.instrument_tests.DummyCam"
+    file = os.path.join(TESTDIR, "fakedata", "dataset_1.yaml")
+    dataIds = [dict(instrument="DummyCam", exposure=100, detector=0)]
+
+    @property
+    def visits(self):
+        butler = Butler(self.root, collections=[self.outputRun])
+        return {
+            DataCoordinate.standardize(
+                instrument="DummyCam",
+                visit=100,
+                universe=butler.registry.dimensions
+            ): [
+                DataCoordinate.standardize(
+                    instrument="DummyCam",
+                    exposure=100,
+                    universe=butler.registry.dimensions
+                )
+            ]
+        }
+
+    def testWriteCuratedCalibrations(self):
+        """There are no curated calibrations in this test instrument"""
+        pass
+
+
+class RawIngestIndexTestCase(RawIngestTestCase):
+    """This time ingest via an index file."""
+    file = os.path.join(TESTDIR, "fakedata2", "dataset_1.yaml")
+
+
+class TestTaskPickle(unittest.TestCase):
```

Comment:
docstring?

Reply:
This was here before and not written by me so I wasn't really sure what to do with it. I'll cobble something together.